sdk/textanalytics/ai-text-analytics/samples/typescript/src/beginAnalyzeBatchActions.ts

```diff
@@ -46,9 +47,9 @@ export async function main() {
     }`
   );
 
-  for await (const page of resultPages) {
-    const keyPhrasesResults = page.extractKeyPhrasesResults![0];
-    for (const doc of keyPhrasesResults) {
+  const keyPhrasesAction = page.extractKeyPhrasesResults[0];
```

Comment:
Just curious what happens when there are no values

Does `const page = (await resultPages.next()).value;` always return an array even if there is no data? That is what I'd expect but figured I'd ask.

The more questionable one here is `page.extractKeyPhrasesResults[0].error` - what if there are no results?

These are just samples / README so not a huge deal, I was just thinking about someone running this sample and getting a `cannot read property 'error' of null` and figured maybe a nicer error would help if they forgot to set something up.

Reply:
> Does const page = (await resultPages.next()).value; always return an array even if there is no data?

No, this is an async iterator where I am dereferencing explicitly here. If there is no data, I will get an undefined but I am sure that I will get data in this case. @bterlson have a nice blog post about them here: https://devblogs.microsoft.com/azure-sdk/async-iterators-in-the-azure-sdk-for-javascript-typescript/

> The more questionable one here is page.extractKeyPhrasesResults[0].error - what if there are no results?

Yeah this could happen if no extract key phrases action was sent. Ok I will update it back to use the for loop.